### ISSUE TITLE
Why is convert.py missing?

### ISSUE THREAD

**Issue by David-AU-github** [ggerganov/llama.cpp#7658]
### What happened?

Critical "non llama3" convert.py and change NOT in download of files.

ALSO:
It is unclear if "convert-hf-to-gguf.py"  supports what "convert.py" did . 

Does it convert llama, llama2, mistral or is "convert-legacy-llama.py" required?
Safetensor files are EVERYWHERE. (?)  [ RE: https://github.com/ggerganov/llama.cpp/pull/7430 ]

This critical action DID NOT OCCUR:
"Move convert.py to examples/convert-legacy-llama.py"

"examples/convert-legacy-llama.py" does not exist. 
(when downloading the zip files).

On another note why remove "convert.py" at all? 

-This breaks "bat files" and automation generation.
-This will break all colabs too.
-This will break any HF spaces that create GGUF files as well.
-This will create needless confusion.

If "convert-hf-to-gguf.py" (llama3) does everything convert.py did , just keep it as "convert.py" ?

### Name and Version

this is not applicable - core files missing.

### Relevant log output

```shell
this is not applicable - core files missing.
```

**Comment by Galunid** [comment]
### TLDR:


`convert-hf-to-gguf.py` is the correct way to convert huggingface models. If you have `llama/llama2` model downloaded directly from meta (in `.pth` format), you can still use `examples/convert-legacy-llama.py`, otherwise it's discourage and we won't provide support. **It will not work with `llama3`**

#### If you need to fix scripts/colabs, changing `convert.py` to `convert-hf-to-gguf.py` and removing `--vocab-type` should be enough to make them work.

---

### Full explanation:

In #7430 `convert.py` was moved to `examples/convert-legacy-llama.py` to avoid confusion as to which script should be used (now there's only one) and reduce maintenance burden on developers. **It should not be used anymore.** It was removed to force people to use `convert-hf-to-gguf.py` that supports all huggingface models that `llama.cpp` can run. There were *a lot* of issues because of people using `convert.py` when they shouldn't have.

I know this will inconvenience some people and it's really unfortunate, but from developer perspective maintaining `convert.py` was just no longer an option. `convert-hf-to-gguf.py` was doing everything `convert.py` did and much more, and it is much simpler to maintain. For a while `convert.py` produced broken models that had warnings like this:

```
llm_load_vocab: missing pre-tokenizer type, using: 'default'
llm_load_vocab:                                             
llm_load_vocab: ************************************        
llm_load_vocab: GENERATION QUALITY WILL BE DEGRADED!        
llm_load_vocab: CONSIDER REGENERATING THE MODEL             
llm_load_vocab: ************************************   
```

Fixes were implemented in `convert-hf-to-gguf.py` that never made it to `convert.py`. Making changes to both scripts that did pretty much the same thing takes a lot of time, so one of them was removed.

If you really need it, it's still available in `examples/convert-legacy-llama.py`, but it's no longer supported. 

>On another note why remove "convert.py" at all?
>
>-This breaks "bat files" and automation generation.
>-This will break all colabs too.
>-This will break any HF spaces that create GGUF files as well.
>-This will create needless confusion.

"Why" is explained above, scripts/colabs fix is to change `convert.py` to `convert-hf-to-gguf.py` and removing unrecognized flags. Our huggingface space should be working as expected. If someone is confused, then most likely they were using the wrong script and their model quality will improve by switching :).

**Comment by David-AU-github** [comment]
Okay - but this DOESN'T EXIST:

examples/convert-legacy-llama.py

It is not in the zip files.

**Comment by David-AU-github** [comment]
> same issue but it worked yesterday so they removed the convert.py a couple of hours ago?
> 
> Problem solved by using CLI command: python llama.cpp/convert-hf-to-gguf.py FOLDER --outfile OUTPUT --outtype f16

thanks - that really helps!!!

**Comment by Galunid** [comment]
> It is not in the zip files.

I'm not sure what you mean. I downloaded the file you linked and it is there.

![image](https://github.com/ggerganov/llama.cpp/assets/10298730/cc94f435-7b0e-4d19-ad5c-97827bc3f01a)

**Comment by mlabonne** [comment]
@Galunid Thanks for this clear explanation, it makes complete sense

**Comment by David-AU-github** [comment]
> > It is not in the zip files.
> 
> I'm not sure what you mean. I downloaded the file you linked and it is there.
> 
> ![image](https://private-user-images.githubusercontent.com/10298730/335511878-cc94f435-7b0e-4d19-ad5c-97827bc3f01a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTcxNjA4NDcsIm5iZiI6MTcxNzE2MDU0NywicGF0aCI6Ii8xMDI5ODczMC8zMzU1MTE4NzgtY2M5NGY0MzUtN2IwZS00ZDE5LWFkNWMtOTc4MjdiYzNmMDFhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTMxVDEzMDIyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMzZjgzODdiYWEwMTAyZDkyMjk4MTc1OTZjYmI1YmFjYjg5MjIwZjlkMzdhMzkwMGNiNjBlZTRkMGNhYWE3NmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ui1I1cZ-muQaK161nSTF-RvlIEUC9HOvz3D5ox41yII)

I will download again - I checked it - not there.
Delay in being added?

**Comment by David-AU-github** [comment]
Okay - there it is . ?(??) ; thanks again.

**Comment by Galunid** [comment]
> There were a lot of issues because of people using convert.py when they shouldn't have.

I'd like to sincerely apologize and clarify I meant [github issues](https://github.com/ggerganov/llama.cpp/issues). It wasn't my intention to imply that users are dumb, problematic or whatever else I may have accidentally implied. All I wanted was merely to show that it was a common problem among the user base and I certainly didn't mean it to be read as a sarcastic remark.

*There were a lot of tickets, where `convert.py` was used by users in situations where it wasn't appropriate.* is much closer to what I meant.

We are aware there was a lack of documentation regarding conversion scripts and it is a big reason why we decided to remove `convert.py` when compilade added partial tensor loading to `convert-hf-to-gguf.py` (hence significantly reducing memory usage of that script).

> There should be a standardization to one script, but the classic lack of communication around here does suck

> As long as all reasonable steps are taken to maintain backwards compatibility, sure. I don't think that's happening here though. Maybe the devs don't understand how integral this project is in the community now? (although, that seems a bit hard to believe). Projects like this and ollama need to start taking their roles more seriously, IMO. I'm sure there's a way to make this transition smoother for users.

> To be clear, this change doesn't affect me. I've been using the new script for a while now. Just trying to bridge the disconnect between devs and normal users

And from other tickets (that as you said "that I've been saying for a while now and taking flak"):
> Thanks for telling us. I gotta say, it's getting real annoying wasting endless hours chasing these things down because the devs can't be bothered to update the relevant info in the main readme (which, BTW, makes no mention of "convert-hf-to-gguf.py" that I'm aware of). Seriously, I can't be the only one who is infuriated by this pattern of behavior in this community.
>
>Devs: documentation matters. What would take you, what, five minutes to update, would save the community probably hundreds if not thousands of cumulative hours. We appreciate what you do (well, I do, anyway), but this is just dumb and lazy. How many botched ggufs are being proliferated because of this?

>Clearly you don't understand. Development is a continuous process and things change quickly here. If you want users to keep up with development and keep the documentation updated, you've skipped CSci 101, where you would have been taught documentation is one of the most important things for a developer to do well. Since when do users write manuals? You'd basically have to become a dev to be able to do that. I don't understand how this is so hard to comprehend. You've also mistaken constructive criticism for complaining. I'm trying to help you devs understand the user perspective. My tone is intentional to convey the frustration many users feel but are too afraid to voice. If you don't see how this is immensely helpful, well, I guess I should have expected that. I don't really care what you think about me. If you want this project to thrive, you need better documentation. Telling the users to create it "isn't productive or helpful."

> *And the rest I haven't bothered to copy over:*

Thank you for your... passionate response.

Let me take this opportunity to explain what constructive criticism is. It's not about throwing around phrases like "classic lack of communication" or accusing others of having an "elitist 'it's the stupid user's fault' attitude". It's about providing specific, actionable feedback that helps improve the project.

Your complaint about documentation being inadequate is valid, and I appreciate you bringing it up. However, instead of attacking us for not doing things your way, why not offer some concrete suggestions on how to improve the documentation? That would be constructive.

As for the removal of the script, I understand it may have caused inconvenience. But let's not assume we're intentionally trying to make life harder for users. We're just trying to avoid confusion and reduce maintenance burden. If you have a better solution, please share it with us.

And yes, projects like ours do need to take their roles more seriously and engage with the community. But that doesn't mean we should be attacked or criticized without any constructive input.

So, let's try to keep the conversation civil and focused on improving the project. We're all in this together, after all.

**Comment by erasmus74** [comment]
> I don't do this because I don't have better things I could be doing.

Go do those things. Save angsty posts for youtube comment section. Let them work.

**Comment by slaren** [comment]
Looks like this has already been answered.
